src/encord_active/lib/encord/project_sync.py

```diff
@@ -226,7 +228,45 @@ def copy_filtered_data(
         label_row["classification_answers"] = {
             k: v for k, v in label_row["classification_answers"].items() if k in filtered_label_hashes
         }
-        target_label_row_structure.set_label_row_json(label_row)
+        label_row_mapping[label_row_hash] = label_row
+
+    with PrismaConnection(curr_project_structure) as conn:
+        all_label_rows = conn.labelrow.find_many(
+            where={
+                "label_hash": {"in": list(filtered_label_rows)},
+            },
+            include={"data_units": {"where": {"data_hash": {"in": list(filtered_data_hashes)}}}},
+        )
+    with PrismaConnection(target_project_structure) as conn:
+        for label_row_db in all_label_rows:
+            conn.labelrow.create(
+                data={
+                    "label_hash": label_row_db.label_hash,
+                    "data_units": {
+                        "create": [
+                            {
+                                "data_hash": data_unit.data_hash,
+                                "data_title": data_unit.data_title,
+                                "frame": data_unit.frame,
+                                "data_uri": local_data_mapping.get(data_unit.data_uri, data_unit.data_uri)
+                                if data_unit.data_uri is not None
+                                else None,
+                                "width": data_unit.width,
+                                "height": data_unit.height,
+                                "fps": data_unit.fps,
+                            }
+                            for data_unit in (label_row_db.data_units or [])
+                        ]
+                    },
+                    "label_row_json": json.dumps(label_row_mapping[label_row_db.label_hash or ""], indent=2),
```

Comment:
[nitpick] 💅

We probably shouldn't add indentation to keep it compact 

```suggestion
                    "label_row_json": json.dumps(label_row_mapping[label_row_db.label_hash or ""]),
```